### PR TITLE
Cant assume effectiveto is there

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Test Steps

New integration tests should pass

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

File: uaclient/jobs/update_messaging.py [hunk 1]
```diff
@@ -71,6 +71,14 @@ def get_contract_expiry_status(
     grace_period = defaults.CONTRACT_EXPIRY_GRACE_PERIOD_DAYS
     pending_expiry = defaults.CONTRACT_EXPIRY_PENDING_DAYS
     remaining_days = cfg.machine_token_file.contract_remaining_days
+
+    # if unknown assume the worst
+    if remaining_days is None:
+        logging.warning(
+            "contract effectiveTo date is null - assuming it is expired"
+        )
+        return ContractExpiryStatus.EXPIRED, -grace_period
```
Comment (lucasmoura) [inline uaclient/jobs/update_messaging.py:80]: @orndorffgrant should we also change that logic for fix too ?
https://github.com/canonical/ubuntu-advantage-client/blob/main/uaclient/security.py#L1292

Just to be consistent here

Comment (orndorffgrant) [reply uaclient/jobs/update_messaging.py:80]: good catch! I'll change it

Comment (orndorffgrant) [reply uaclient/jobs/update_messaging.py:80]: pushed!

File: uaclient/security.py [hunk 2]
```diff
@@ -1270,13 +1270,13 @@ def _check_subscription_is_expired(
...
     contract_expiry_datetime = status_cache.get("expires")
-    # If we don't have an expire information on the status-cache, we
-    # can assume that the machine is not attached.
-    if contract_expiry_datetime is None:
-        return False
 
-    tzinfo = contract_expiry_datetime.tzinfo
-    if contract_expiry_datetime < datetime.now(tzinfo):
+    # If we don't have an expire information on the status-cache, we
+    # assume that the machine is not attached.
```
Comment (lucasmoura) [inline uaclient/security.py:1275]: I think we can also remove this comment here

**Review by lucasmoura — APPROVED** [review]
Just one minor nit

**Review by lucasmoura — APPROVED** [review]

**Comment by orndorffgrant** [comment]
oops I'll fix the failures

**Pull request merged**